### PR TITLE
Add PNG download and white background for heatmaps

### PR DESCRIPTION
**Pull request by gregdingle**
# Description

image

As title. I learned that PNG is supported by the existing lib, but not jpg, which I think is fine. 

Also, I think I fixed the root cause of the SVG "compatibility" issues. Before they were downloaded with a transparent background which may appear as gray in some applications. I changed the background to white for the heatmap globally which I think is appropriate. If we ever want to change the page color background, it will be obvious what to do. 

# Test and Reproduce

1. click "download png"
2. see download
3. open download and see good png on white background
4. repeat for svg download

### CODE REVIEW

File: app/assets/src/components/views/compare/SamplesHeatmapView.jsx [hunk 1]
```diff
@@ -690,8 +690,13 @@ class SamplesHeatmapView extends React.Component {
   handleDownloadClick = fileType => {
     switch (fileType) {
       case "svg":
+        // TODO (gdingle): pass in filename per sample?
         this.heatmapVis.download();
         break;
+      case "png":
+        // TODO (gdingle): pass in filename per sample?
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/compare/SamplesHeatmapView.jsx]: What do you mean by these?

File: app/assets/src/components/visualizations/heatmap/Heatmap.js [hunk 2]
```diff
@@ -56,7 +56,9 @@ export default class Heatmap {
         columnMetadata: [],
         metadataColorScale: new CategoricalColormap(),
         enableColumnMetadata: false,
-        iconPath: "/assets/icons"
+        iconPath: "/assets/icons",
+        // This is needed for downloading PNG and SVG on solid background
+        backgroundColor: "white"
```
Comment (tfrcarvalho) [inline app/assets/src/components/visualizations/heatmap/Heatmap.js]: Can you call this `svgBackgroundColor` ?

File: app/assets/src/components/visualizations/heatmap/Heatmap.js [hunk 3]
```diff
@@ -204,7 +206,10 @@ export default class Heatmap {
       .append("svg")
       .attr("class", cs.heatmap)
       .attr("id", "visualization")
-      .attr("xmlns", "http://www.w3.org/2000/svg");
+      .attr("xmlns", "http://www.w3.org/2000/svg")
+      // Not standard but it works for downloads and svgsaver. See:
+      // https://stackoverflow.com/questions/11293026/default-background-color-of-svg-root-element
+      .attr("style", "background-color: " + this.options.backgroundColor);
```
Comment (tfrcarvalho) [inline app/assets/src/components/visualizations/heatmap/Heatmap.js]: Use interpolation here: `background-color: ${this.options.backgroundColor}`

**Comment by jshoe** [comment]
PNG looks great although it still has the black background in my Illustrator. You could ask IT for a Creative Cloud account perhaps

Screen Shot 2019-03-12 at 2 06 12 PM

File: app/assets/src/components/views/compare/SamplesHeatmapView.jsx (hunk 1 above)
Comment (gregdingle) [reply app/assets/src/components/views/compare/SamplesHeatmapView.jsx]: Make the filename unique for the set of sample selected.

**Comment by gregdingle** [comment]
> PNG looks great although it still has the black background in my Illustrator. You could ask IT for a Creative Cloud account perhaps. not blocking

@jshoe thanks for checking. Are you sure it was the PNG? It says "svg" in your screenshot.

**Comment by jshoe** [comment]
@gregdingle sorry, meant to say "the SVG still has the black background".

File: app/assets/src/components/views/compare/SamplesHeatmapView.jsx (hunk 1 above)
Comment (tfrcarvalho) [reply app/assets/src/components/views/compare/SamplesHeatmapView.jsx]: I would not add comments to the code for things that we *might* consider doing... a JIRA task might be better for that... 
If you still keep the comment, at least change to "per sample set".

Comment (gregdingle) [reply app/assets/src/components/views/compare/SamplesHeatmapView.jsx]: Yes, I put some of these comments in for discussion in review. I'll make this one a low pri JIRA task.

**Pull request merged**